mlir/utils/jenkins/MIOpenDriver.py

```diff
@@ -11,29 +12,91 @@
 import math
 import itertools
 from datetime import date
+from pathlib import Path
+import glob
+import argparse
 
 import numpy as np
 import pandas as pd
 import re
 
 # global variables.
-MLIR_BIN_DIR = './bin'
-LLVM_BIN_DIR = './external/llvm-project/llvm/bin'
-MIOPEN_GEN = 'miopen-gen'
-MLIR_MIOPEN_DRIVER = 'mlir-miopen-driver'
-MLIR_ROCM_RUNNER = 'mlir-rocm-runner'
-MLIR_ROCM_RUNNER_ARGS = ['--shared-libs=./external/llvm-project/llvm/lib/libmlir_rocm_runtime.so,./lib/libconv-validation-wrappers.so,./external/llvm-project/llvm/lib/libmlir_runner_utils.so', '--entry-point-result=void']
 ROCPROF = '/opt/rocm/bin/rocprof'
-MIOPEN_DRIVER = '../MIOpen/build/bin/MIOpenDriver'
 BENCHMARKING_RESULT_FILE_NAME = 'results.stats.csv'
-CONFIGURATION_FILE_NAME ='../mlir/utils/jenkins/miopen-tests/resnet50-miopen-configs'
-
 DIRECTIONS = ['-F 1', '-F 2', '-F 4']
 DATA_TYPES = ['conv', 'convfp16', 'convint8']
 LAYOUTS = ['NHWC', 'NCHW']
+
 # Compiled regexp object used for extracting elapsed time from MIOpenDriver's output
 ELAPSED_TIME_RE = re.compile(r"Elapsed: (.*)ms")
 
+class Paths(NamedTuple):
+    """This structure is used to hold paths needed to perform the tests"""
+    miopen_gen_path : str
```

Comment:
Can we make this a `Path` object or, failing that, `bytes`? Paths aren't necessarily UTF-anything.

Reply:
we can -- it is just the consumers of this has to convert them to strings to be used by sub process.
Do you see a benefit in doing so ?

Reply:
If `subprocess` will only take strings, then, yeah, ignore all that and use strings. However, I think you can pass a `Path` subclass to the subprocesses runner, but it might take `shell=False`

Reply:
Having thought about this more, I feel keeping them in strings is much useful.
It is true that Path object can be passed onto subprocess but there are not passed as it is -- there are additional arguments concat'd to them.
Moreover, there library paths here that will form into composite arguments such as "--shared-libs".

Therefore, to keep consistency, I feel keeping them string is a better choice.